src/helm/benchmark/adapter.py

```diff
@@ -902,6 +903,22 @@ class labels.
         sorted_labels: List[str] = [
             key for key, _ in sorted(label_to_instances.items(), key=lambda x: len(x[1]), reverse=True)
         ]
+
+        # Break ties by randomly shuffling labels that have the same number of Instances
```

Comment:
I thought what you had before was easier to reason about - I think we can make this logic more functional (rather than procedural) by shuffling the instances for each count, and then concatenating all the lists at the end.

Reply:
Ah yes, that's much cleaner.